frontend/src/scenes/plugins/edit/PluginDrawer.tsx

```diff
@@ -75,14 +77,20 @@ export function PluginDrawer(): JSX.Element {
             form.setFieldsValue({
                 ...(editingPlugin.pluginConfig.config || defaultConfigForPlugin(editingPlugin)),
                 __enabled: editingPlugin.pluginConfig.enabled,
+                __matchActionId: editingPlugin.pluginConfig.match_action,
                 ...editingPluginInitialChanges,
             })
             generateApiKeysIfNeeded(form)
         } else {
             form.resetFields()
         }
         updateInvisibleAndRequiredFields()
-    }, [editingPlugin?.id, editingPlugin?.config_schema])
+    }, [editingPlugin?.id, editingPlugin?.config_schema, editingPlugin?.pluginConfig?.match_action])
+
+    const actionMatchingFlag = !!useFeatureFlag('PLUGINS_ACTION_MATCHING')
```

Comment:
move these changes to pipeline 3000

Reply:
Yeah it wasn't ready hence didn't do it there yet.